### PR TITLE
Add credo linter

### PR DESCRIPTION
**Pull request by skbolton**
This is the standard linter used in the elixir community.

### CODE REVIEW

**Comment by skbolton** [comment]
Just need to add tests, sorry Github is not allowing me to convert this pr to a draft

**Comment by skbolton** [comment]
Left a comment on #72 not realizing there was a PR already. I am willing to take the best of both and get this across the line or let the original pr author work through it if they are responsive.

**Comment by skbolton** [comment]
All fixed up and ready to go @mfussenegger

File: tests/credo_spec.lua [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      end_col = 10,
+      lnum = 0,
+      end_lnum = 0,
+      severity = 1,
+      message = 'Unless conditions should avoid having an `else` block.'
+    }
+
+    assert.are.same(expected_error, result[1])
+
+    local expected_error = {
```
Comment (mfussenegger) [inline tests/credo_spec.lua:24]:
```suggestion
    expected_error = {
```

File: lua/lint/linters/credo.lua [hunk 2]
```diff
@@ -0,0 +1,12 @@
+local errorfmt = '[%t] %. %f:%l:%c %m, [%t] %. %f:%l %m'
+
+return {
```
Comment (mfussenegger) [inline lua/lint/linters/credo.lua:3]:
```suggestion
return {
```

**Pull request merged**